src/core/thread/router_table.cpp

```diff
@@ -564,6 +564,7 @@ void RouterTable::ProcessTlv(const ThreadRouterMaskTlv &aTlv)
         else
         {
             mAllocatedRouterIds.Remove(routerId);
+            Get<AddressResolver>().Remove(routerId);
```

Comment:
Can we even remove the neighbor earlier?

Reply:
Existing implementation has already updated router table accordingly, see https://github.com/openthread/openthread/blob/master/src/core/thread/router_table.cpp#L539

Reply:
Why is the logic here not exactly same as in function `void RouterTable::ProcessTlv(const Mle::RouteTlv &aTlv)` ?